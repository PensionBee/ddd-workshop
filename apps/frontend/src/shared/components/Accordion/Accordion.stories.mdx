import { Meta, Story, Canvas, ArgsTable } from "@storybook/addon-docs";
import { Accordion } from "./Accordion";

<Meta
  title="Shared/Accordion/Docs"
  component={Accordion}
  parameters={{
    viewMode: "docs",
    previewTabs: {
      canvas: { hidden: true },
    },
  }}
/>

# Accordion

## Props

<ArgsTable of={Accordion} />

## Usage & Customisation

### Default

<Canvas>
  <Story id="shared-accordion--default" />
</Canvas>

### Initially Open

<Canvas>
  <Story id="shared-accordion--initially-open" />
</Canvas>

### With Links

<Canvas>
  <Story id="shared-accordion--with-link" />
</Canvas>
