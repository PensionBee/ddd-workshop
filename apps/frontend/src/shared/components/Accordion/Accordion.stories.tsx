import { Box } from "@shared/components/Box";
import { Link } from "@shared/components/Link";
import { BodySmallL } from "@shared/components/Typography";
import type { ComponentStory, ComponentMeta } from "@storybook/react";
import type { AccordionProps } from "./Accordion";
import { Accordion } from "./Accordion";

const loremIpsum =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Et tincidunt faucibus neque scelerisque sagittis. Vitae nibh congue placerat enim tortor vitae dictum eu mi. Faucibus tincidunt rhoncus turpis eget.";

export default {
  title: "Shared/Accordion",
  component: Accordion,
  parameters: {
    viewMode: "story",
    previewTabs: {
      "storybook/docs/panel": { hidden: true },
    },
  },
} as ComponentMeta<typeof Accordion>;

const AccordionTemplate: ComponentStory<typeof Accordion> = (
  args: AccordionProps
) => {
  return <Accordion {...args} />;
};

export const Default = AccordionTemplate.bind({});
Default.args = {
  items: [
    {
      id: "first-item",
      heading: "First Item",
      content: loremIpsum,
    },
    {
      id: "second-item",
      heading: "Second Item",
      content: loremIpsum,
    },
    {
      id: "third-item",
      heading: "Third Item",
      content: loremIpsum,
    },
  ],
};

export const InitiallyOpen = AccordionTemplate.bind({});
InitiallyOpen.args = {
  items: [
    {
      id: "first-item",
      heading: "First Item",
      content: loremIpsum,
    },
    {
      id: "second-item",
      heading: "Second Item",
      content: loremIpsum,
    },
    {
      id: "third-item",
      heading: "Third Item",
      content: loremIpsum,
    },
  ],
  isInitiallyOpen: true,
};

const LinkContent = () => (
  <Box d="flex" direction="column">
    <BodySmallL>
      {loremIpsum} <Link href="/">An inline link</Link>
    </BodySmallL>
    <Box mb={24} />
    <Link href="/">A longer full-width link.</Link>
  </Box>
);

export const WithLink = AccordionTemplate.bind({});
WithLink.args = {
  items: [
    {
      id: "first-item",
      heading: "First Item",
      content: <LinkContent />,
    },
    {
      id: "second-item",
      heading: "Second Item",
      content: <LinkContent />,
    },
    {
      id: "third-item",
      heading: "Third Item",
      content: <LinkContent />,
    },
  ],
};
