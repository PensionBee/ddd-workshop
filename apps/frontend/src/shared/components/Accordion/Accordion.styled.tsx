import * as RadixAccordion from "@radix-ui/react-accordion";
import { theme } from "@shared/styling/theme";
import * as typography from "@shared/styling/typography";
import styled from "styled-components";

export const StyledAccordion = styled(RadixAccordion.Root)`
  background: ${theme.colors.white};
  padding: 8px;
  @keyframes accordion__opening {
    0% {
      padding-top: 0;
      height: 0;
    }
    100% {
      padding-top: 20px;
      height: var(--radix-accordion-content-height);
    }
  }
  @keyframes accordion__closing {
    0% {
      padding-top: 20px;
      height: var(--radix-accordion-content-height);
    }
    100% {
      padding-top: 0;
      height: 0;
    }
  }
`;

export const StyledItem = styled(RadixAccordion.Item)`
  margin: 0 0 28px 0;
`;

export const StyledTrigger = styled(RadixAccordion.Trigger)`
  width: 100%;
  cursor: pointer;
  border: 0px;
  display: flex;
  align-items: center;
  &:focus-visible {
    outline: 3px solid ${theme.colors.blue100};
  }
  &:focus-visible [data-state="open"] {
    outline: 3px solid ${theme.colors.blue100};
  }
`;

export const StyledContent = styled(RadixAccordion.Content)`
  padding-left: 40px;
  padding-top: 20px;
  overflow: hidden;
  ${() => typography.bodySmallL}
  ${`color: ${theme.colors.black};`}
  &[data-state="open"] {
    animation: accordion__opening 500ms;
  }
  &[data-state="closed"] {
    animation: accordion__closing 500ms;
  }
  // Provide enough space for focus outlines
  padding-right: 4px;
  padding-bottom: 4px;
  margin-right: -4px;
  margin-bottom: -4px;
`;

export const StyledHeading = styled.div`
  transition: color 500ms ease;
  ${() => typography.bodySmallB}
  ${`color: ${theme.colors.black};`}
  [data-state="open"] & {
    ${`color: ${theme.colors.grey1};`}
  }
  &:hover {
    ${`color: ${theme.colors.grey1};`}
  }
`;
