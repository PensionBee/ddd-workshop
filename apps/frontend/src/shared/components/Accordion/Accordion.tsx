import React, { useState } from "react";
import * as RadixAccordion from "@radix-ui/react-accordion";
import {
  StyledAccordion,
  StyledContent,
  StyledHeading,
  StyledItem,
  StyledTrigger,
} from "./Accordion.styled";
import { LargeSystemIcon } from "../Svg/Svg";

export type AccordionItem = {
  id: string;
  heading: React.ReactNode;
  content: React.ReactNode;
};

export type AccordionProps = {
  /** A list of accordion items */
  items: AccordionItem[];
  /** Whether the accordion should initially be open */
  isInitiallyOpen?: boolean;
};

export const Accordion: React.FC<AccordionProps> = ({
  items,
  isInitiallyOpen = false,
  ...props
}) => {
  const [activeItem, setActiveItem] = useState(
    isInitiallyOpen ? items[0].id : null
  );

  const handleValueChange = (value: string) => {
    return setActiveItem(value);
  };

  return (
    <StyledAccordion
      data-rc="Accordion"
      collapsible
      type="single"
      defaultValue={isInitiallyOpen ? items[0].id : undefined}
      onValueChange={handleValueChange}
      {...props}
    >
      {items.map((item) => {
        return (
          <StyledItem value={item.id} key={item.id}>
            <RadixAccordion.Header>
              <StyledTrigger>
                <LargeSystemIcon
                  icon={activeItem === item.id ? "MinusYellow" : "PlusYellow"}
                  w={16}
                  h={16}
                  m={2}
                  mr={20}
                />
                <StyledHeading>{item.heading}</StyledHeading>
              </StyledTrigger>
            </RadixAccordion.Header>
            <StyledContent>{item.content}</StyledContent>
          </StyledItem>
        );
      })}
    </StyledAccordion>
  );
};
