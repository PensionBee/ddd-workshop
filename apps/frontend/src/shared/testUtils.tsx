import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
