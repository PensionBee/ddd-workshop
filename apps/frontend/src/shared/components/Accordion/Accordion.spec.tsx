import { fireEvent, render } from "@shared/testUtils";
import { Accordion } from "./Accordion";

const accordionItems = [
  {
    id: "item-1",
    heading: "Heading 1",
    content: "Content 1",
  },
  {
    id: "item-2",
    heading: "Heading 2",
    content: "Content 2",
  },
  {
    id: "item-3",
    heading: "Heading 3",
    content: "Content 3",
  },
];

describe("Default Accordion", () => {
  it("should render correctly", () => {
    const { queryByText, getByText, asFragment } = render(
      <Accordion items={accordionItems} />
    );

    // Check that the correct elements are rendered/visible
    accordionItems.forEach((item) => {
      expect(getByText(item.heading)).toBeVisible();
      expect(queryByText(item.content)).toBeNull();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("should show the correct dropdown content when a heading is clicked", () => {
    const { queryByText, getByText, asFragment } = render(
      <Accordion items={accordionItems} />
    );

    // Open the first item
    const firstItemHeading = getByText(accordionItems[0].heading);
    fireEvent.click(firstItemHeading);

    // Check that only the first item content is visible
    expect(getByText(accordionItems[0].content)).toBeVisible();
    expect(queryByText(accordionItems[1].content)).toBeNull();
    expect(queryByText(accordionItems[2].content)).toBeNull();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should close open content when another heading is clicked", () => {
    const { queryByText, getByText, asFragment } = render(
      <Accordion items={accordionItems} />
    );

    // Open the first item
    const firstItemHeading = getByText(accordionItems[0].heading);
    fireEvent.click(firstItemHeading);

    // Check that only the first item content is visible
    expect(getByText(accordionItems[0].content)).toBeVisible();
    expect(queryByText(accordionItems[1].content)).toBeNull();
    expect(queryByText(accordionItems[2].content)).toBeNull();

    // Open the second item
    const secondItemHeading = getByText(accordionItems[1].heading);
    fireEvent.click(secondItemHeading);

    // Check that only the second item content is visible
    expect(queryByText(accordionItems[0].content)).toBeNull();
    expect(getByText(accordionItems[1].content)).toBeVisible();
    expect(queryByText(accordionItems[2].content)).toBeNull();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should close open content when an open item's heading is clicked again", () => {
    const { queryByText, getByText, asFragment } = render(
      <Accordion items={accordionItems} />
    );

    // Open the third item
    const thirdItemHeading = getByText(accordionItems[2].heading);
    fireEvent.click(thirdItemHeading);

    // Check that only the third item content is visible
    expect(queryByText(accordionItems[0].content)).toBeNull();
    expect(queryByText(accordionItems[1].content)).toBeNull();
    expect(getByText(accordionItems[2].content)).toBeVisible();

    // Close the third item
    fireEvent.click(thirdItemHeading);

    // Check that all content is hidden
    accordionItems.map((item) => {
      return expect(queryByText(item.content)).toBeNull();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Initially Open Accordion", () => {
  it("should show the first item content on render", () => {
    const { queryByText, getByText, asFragment } = render(
      <Accordion isInitiallyOpen items={accordionItems} />
    );

    expect(getByText(accordionItems[0].content)).toBeVisible();
    expect(queryByText(accordionItems[1].content)).toBeNull();
    expect(queryByText(accordionItems[2].content)).toBeNull();

    expect(asFragment()).toMatchSnapshot();
  });
});
