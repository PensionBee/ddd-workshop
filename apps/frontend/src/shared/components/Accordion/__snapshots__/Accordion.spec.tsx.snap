// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Accordion should close open content when an open item's heading is clicked again 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bSmHjB"
    data-rc="Accordion"
  >
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rj:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:ri:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 1
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:ri:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rj:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rl:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rk:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 2
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rk:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rl:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rn:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rm:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 3
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rm:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rn:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Default Accordion should close open content when another heading is clicked 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bSmHjB"
    data-rc="Accordion"
  >
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rd:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rc:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 1
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rc:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rd:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="open"
    >
      <h3
        data-state="open"
      >
        <button
          aria-controls="radix-:rf:"
          aria-expanded="true"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="open"
          id="radix-:re:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.MinusYellow"
          >
            <svg>
              minus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 2
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:re:"
        class="sc-hLBbgP lbVTL"
        data-state="open"
        id="radix-:rf:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      >
        Content 2
      </div>
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rh:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rg:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 3
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rg:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rh:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Default Accordion should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bSmHjB"
    data-rc="Accordion"
  >
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 1
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r0:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:r1:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r2:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 2
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:r5:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r4:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 3
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r4:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:r5:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Default Accordion should show the correct dropdown content when a heading is clicked 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bSmHjB"
    data-rc="Accordion"
  >
    <div
      class="sc-gswNZR eoZMGq"
      data-state="open"
    >
      <h3
        data-state="open"
      >
        <button
          aria-controls="radix-:r7:"
          aria-expanded="true"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="open"
          id="radix-:r6:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.MinusYellow"
          >
            <svg>
              minus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 1
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r6:"
        class="sc-hLBbgP lbVTL"
        data-state="open"
        id="radix-:r7:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      >
        Content 1
      </div>
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:r9:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r8:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 2
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r8:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:r9:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rb:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:ra:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 3
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:ra:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rb:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Initially Open Accordion should show the first item content on render 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe bSmHjB"
    data-rc="Accordion"
  >
    <div
      class="sc-gswNZR eoZMGq"
      data-state="open"
    >
      <h3
        data-state="open"
      >
        <button
          aria-controls="radix-:rp:"
          aria-expanded="true"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="open"
          id="radix-:ro:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.MinusYellow"
          >
            <svg>
              minus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 1
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:ro:"
        class="sc-hLBbgP lbVTL"
        data-state="open"
        id="radix-:rp:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
      >
        Content 1
      </div>
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rr:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rq:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 2
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rq:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rr:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
    <div
      class="sc-gswNZR eoZMGq"
      data-state="closed"
    >
      <h3
        data-state="closed"
      >
        <button
          aria-controls="radix-:rt:"
          aria-expanded="false"
          class="sc-dkrFOg fYgpNG"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:rs:"
          type="button"
        >
          <span
            class="sc-jSUZER TGmAq"
            data-rc="LargeSystemIcon.PlusYellow"
          >
            <svg>
              plus-yellow.svg
            </svg>
          </span>
          <div
            class="sc-eDvSVe eErMbn"
          >
            Heading 3
          </div>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:rs:"
        class="sc-hLBbgP lbVTL"
        data-state="closed"
        hidden=""
        id="radix-:rt:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</DocumentFragment>
`;
